src/mono/mono/mini/aot-compiler.c

```diff
@@ -8629,7 +8652,8 @@ mono_aot_parse_options (const char *aot_options, MonoAotOptions *opts)
 			printf ("    bitcode\n");
 			printf ("    data-outfile=\n");
 			printf ("    direct-icalls\n");
-			printf ("    direct-pinvoke\n");
```

Comment:
We need to keep this or we break command line combability, so old way should work meaning direct-pinvoke for all dllimports as well as the new once, it should not be possible to combine the old with the new arguments though.

Reply:
Reverted to retain the `direct-pinvoke` argument. I reordered the arguments in `mono_aot_parse_options` so `direct-pinvokes=` and `direct-pinvoke-lists=` dont get mistaken for `direct-pinvoke`.
Moreover, I made it such that `direct-pinvoke` boolean flag paths trump the `direct-pinvokes` and `direct-pinvoke-lists` paths.

> it should not be possible to combine the old with the new arguments though
Could you elaborate more on this?

Reply:
If user uses direct-pinvoke it means all dllimports should be direct pinvoked, so using that with the new arguments doesn't make sense and should give an error, direct-pinvoke should have precedence over the other arguments.

Reply:
the `direct-pinvoke` flag should have precedence over the other `direct-pinvokes` and `direct-pinvoke-lists` arguments now, but do we need to throw an error? Currently there is a 
```
if ((acfg->aot_opts.direct_pinvoke && (acfg->aot_opts.direct_pinvokes || acfg->aot_opts.direct_pinvoke_list))
		g_warning ("The 'direct-pinvoke' argument trumps specified 'direct-pinvokes' and 'direct-pinvoke-lists' arguments.\n");
```

Reply:
Since its a configuration error, like using direct-pinvoke without static link I believe we should do the same thing, making sure the user adjust the parameters correctly.

Reply:
Changed it to throw an error instead now